scripts/check-swift-format.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftNIO open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+set -euo pipefail
+
+log() { printf -- "** %s\n" "$*" >&2; }
+error() { printf -- "** ERROR: %s\n" "$*" >&2; }
+fatal() { error "$@"; exit 1; }
+
+excluded_files=""
+
+if [ -f .swiftformatignore ]; then
+    log "Found swiftformatignore file..."
+    excluded_files=$(cat .swiftformatignore | xargs -I% printf ":(exclude)% ")
+fi
+
+log "Running swift format format..."
+git ls-files -z '*.swift' $excluded_files | xargs -0 swift format format --parallel --in-place
+
+log "Running swift format lint..."
+
+git ls-files -z '*.swift' $excluded_files | xargs -0 swift format lint --strict --parallel
+
+log "Checking for modified files..."
+
+GIT_PAGER= git diff --exit-code '*.swift'
```

Comment:
No I set the `GIT_PAGE` to null intentionally.

Reply:
gotcha. Might be nice to be explicit here: `GIT_PAGER='' git diff ...`

Reply:
This is very idiomatic shell already.

If we're gonna make it explicit, I'd rather `env -u GIT_PAGER git diff ...`.

Reply:
IMHO that's obfuscation. I'd suggest to ask a few people what happens if you do

```
FOO= hello world
```

vs.

```
FOO='' hello world
```

In my mind, the first one is confusing, it might be an assignment of `hello world` to `FOO=` or it might be starting `hello` with parameter `world` and env var `FOO=''`.

The second one is obvious.

Reply:
Did you mean the use of `env` is obfuscation? If anything it's clearer and has different semantics. It _unsets_ the variable, as opposed to setting it to the empty string. 

I concede that for setting a variable to the empty string, `FOO=''` is clearer than `FOO=`.

I don't think it's quite as much of an alarm bell as you make out, and `shellcheck` doesn't complain, but I'm happy for it to change if it irks you.

Reply:
Yes, I was referring to `env`. If this is common knowledge that `env -u FOO hello world` runs `hello world`, then I'm completely happy with that of course. As am I with

```
unset FOO
hello world
```

or

```
FOO='' hello world
```

because I think they convey the meaning pretty well.

But out of all the issues, this really is the most minuscule one. Feel free to leave `GIT_PAGER= git diff`, it just reads a little off in my head at least.